optimum/intel/openvino/modeling_decoder.py

```diff
@@ -651,6 +764,954 @@ def _from_pretrained(
 
         return causal_model
 
+    def _beam_search(
```

Comment:
Could you specify using comments which part has been modified from the original code, this would help a lot for review and future maintenance

Reply:
+1